src/components/DatePickerRange.react.js

```diff
@@ -51,18 +51,19 @@ export default class DatePickerRange extends Component {
     }
     onDatesChange({startDate: start_date, endDate: end_date}) {
         const {setProps, updatemode} = this.props;
+
         const old_start_date = this.state.start_date;
         const old_end_date = this.state.end_date;
-        const newState = {};
-        if (setProps && start_date !== null && start_date !== old_start_date) {
+
+        this.setState({start_date, end_date});
+
+        if (start_date && !start_date.isSame(old_start_date)) {
```

Comment:
.isSame, .isBefore, .isAfter are functions offered by moment to compare moment objects to various time formats